medicines/doc-index-updater/src/pars_upload.rs

```diff
@@ -69,49 +70,53 @@ fn document_from_form_data(storage_file: StorageFile, metadata: BlobMetadata) ->
 }
 
 async fn queue_pars_upload(
-    job_id: Uuid,
     form_data: FormData,
     state_manager: impl JobStatusClient,
-) -> Result<(), Rejection> {
+) -> Result<Vec<Uuid>, Rejection> {
     let (metadatas, file_data) = read_pars_upload(form_data).await.map_err(|e| {
         tracing::debug!("Error reading PARS upload: {:?}", e);
         warp::reject::custom(e)
     })?;
 
+    let mut job_ids = Vec::with_capacity(metadatas.len());
+
     for metadata in metadatas {
-        let storage_file = add_file_to_temporary_blob_storage(job_id, &file_data)
-            .await
-            .map_err(warp::reject::custom)?;
+        let job_id = accept_job(&state_manager).await?.id;
+
+        job_ids.push(job_id);
+
+        let storage_file =
+            add_file_to_temporary_blob_storage(job_id, &file_data, &metadata.pl_number)
+                .await
+                .map_err(warp::reject::custom)?;
 
         let document = document_from_form_data(storage_file, metadata);
 
         check_in_document_handler(document, &state_manager).await?;
     }
 
-    Ok(())
+    Ok(job_ids)
 }
 
 async fn upload_pars_handler(
     form_data: FormData,
     state_manager: StateManager,
 ) -> Result<impl Reply, Rejection> {
-    tracing::debug!("Received PARS submission");
-
-    let job_id = accept_job(&state_manager).await?.id;
+    let request_id = Uuid::new_v4();
 
-    let span = tracing::info_span!("Queueing PARS upload", job_id = job_id.to_string().as_str());
+    let span = tracing::info_span!("PARS upload", request_id = request_id.to_string().as_str());
     let _enter = span.enter();
 
-    queue_pars_upload(job_id, form_data, state_manager).await?;
+    tracing::debug!("Received PARS submission");
+
+    let job_ids = queue_pars_upload(form_data, state_manager).await?;
```

Comment:
This is better now that we're returning all the different job ids, however adding to the queue is still synchronous, so the user has to wait for _n_ temporary blob uploads before receiving feedback on submission.